vignettes/sass.Rmd

```diff
@@ -1,165 +1,402 @@
 
-h1 {
-  color: font-color($color);
-}
+### Mixins {#mixins}
 
-.shiny-plot-output {
-  max-width: percentage($width / 100);
-}
+Similar to how functions are useful for encapsulating _computation_ in a reusable unit, [mixins](https://sass-lang.com/documentation/at-rules/mixin) are useful for doing the same with _styling rules_ (i.e., packaging them into a reusable unit).
+Technically speaking, mixins are similar to functions in that they require a `name`, may have arguments, as well as any number of statements. However, they differ in that they require the return value to be a [style rule](https://sass-lang.com/documentation/style-rules), and when called, need to be `@include`d in a larger style rule in order to generate any CSS. 
+
+For some examples, please see the [Sass mixin documentation](https://sass-lang.com/documentation/at-rules/mixin).
+
+### More basics
+
+This vignette intentionally doesn't try to re-invent the existing and wonderful [Sass documentation](https://sass-lang.com/documentation). There you'll find many more useful things as you start to write more Sass, such as [control flow](https://sass-lang.com/documentation/at-rules/control), [lists](https://sass-lang.com/documentation/values/lists), [maps](https://sass-lang.com/documentation/values/maps), [interpolation](https://sass-lang.com/documentation/interpolation), and more.
+
+## Composable sass {#layering}
+
+To make Sass code more composable with other Sass code (e.g., allowing others to change your variable defaults or import a function or mixin you've defined), consider partitioning your Sass code into a `sass::sass_layer()`. The main idea is to split your Sass into 3 parts: `defaults` (i.e. variable defaults), `declarations` (i.e., [functions](#functions), [mixins](#mixins), etc), and `rules` (i.e., styling rules).
+
+```{r}
+layer1 <- sass_layer(
+  defaults = list("body-bg" = "white !default"),
+  declarations = sass_file("color-contrast.scss"),
+  rules = "body{background-color: $body-bg; color: color-contrast($body-bg)}"
+)
+sass(layer1)
 ```
 
-```{r echo = FALSE}
-knitr::include_app("https://gallery.shinyapps.io/sass-size/", height = "600px")
+This allows downstream `sass_layer()`s to be `sass_layer_merge()`d into a single layer, where those downstream layers are granted higher priority. More specifically, this means: 
+
+* `defaults` for `layer2` are placed _before_ `defaults` for `layer1`.
+  * Allowing downstream Sass to override variable defaults in upstream Sass.
+* `declarations` for `layer2` are placed _after_ `declarations` for `layer1`.
+  * Allowing downstream Sass to use functions and mixins defined in upstream Sass.
+* `rules` for `layer2` are placed _after_ `rules` for `layer1`.
+  * Allowing downstream Sass rules to override upstream rules.
```

Comment:
I'd change "override" to "take precedence over" (and maybe point out that in CSS, if two selectors have otherwise-identical priority (or I guess the technically correct term is "specificity"), the later one wins).